app/src/main/kotlin/com/softteco/template/data/profile/ProfileRepositoryImpl.kt

```diff
@@ -1,44 +1,39 @@
 package com.softteco.template.data.profile
 
 import androidx.datastore.core.DataStore
-import androidx.datastore.preferences.core.Preferences
-import androidx.datastore.preferences.core.edit
-import androidx.datastore.preferences.core.stringPreferencesKey
 import com.softteco.template.data.RestCountriesApi
 import com.softteco.template.data.TemplateApi
 import com.softteco.template.data.base.error.ErrorEntity
 import com.softteco.template.data.base.error.ErrorHandler
 import com.softteco.template.data.base.error.Result
+import com.softteco.template.data.profile.dto.AuthTokenDto
 import com.softteco.template.data.profile.dto.CreateUserDto
 import com.softteco.template.data.profile.dto.CredentialsDto
 import com.softteco.template.data.profile.dto.NewPasswordDto
+import com.softteco.template.data.profile.dto.ProfileDto
 import com.softteco.template.data.profile.dto.ResetPasswordDto
 import com.softteco.template.data.profile.dto.toModel
 import com.softteco.template.data.profile.entity.AuthToken
 import com.softteco.template.data.profile.entity.Profile
 import com.softteco.template.data.profile.entity.Profile.Companion.toJson
-import com.softteco.template.utils.getFromDataStore
-import com.softteco.template.utils.saveToDataStore
 import kotlinx.coroutines.flow.first
 import javax.inject.Inject
+import javax.inject.Named
 import javax.inject.Singleton
 
-private const val AUTH_TOKEN_KEY = "auth_token"
-private const val PROFILE_KEY = "profile"
-
 @Singleton
 internal class ProfileRepositoryImpl @Inject constructor(
     private val templateApi: TemplateApi,
     private val errorHandler: ErrorHandler,
-    private val dataStore: DataStore<Preferences>,
+    @Named("authToken") private val authTokenEncryptedDataStore: DataStore<AuthTokenDto>,
```

Comment:
After removing `authToken` naming from the DataStore module, `@Named("authToken")` here also will be not needed.